src/gui/mzroll/pollyelmaveninterface.cpp

```diff
@@ -927,6 +924,7 @@ QStringList PollyElmavenInterfaceDialog::_prepareFilesToUpload(QDir qdir,
         QString sampleCohortFileName = _writeableTempDir + QDir::separator() + datetimestamp +
                                         "_Cohort_Mapping_Elmaven.csv";
         _mainwindow->projectDockWidget->prepareSampleCohortFile(sampleCohortFileName);
+        _lastCohortFileWasValid = _pollyIntegration->validSampleCohort(CohortFileName);
```

Comment:
Could you reset this variable to `false` after every logout or every time an upload process is complete?

I can't think of any point where this might fail, since this is only used in PollyPhi, but suggesting this as a fail-safe.

Reply:
Right, that makes sense. No flags from a previous session should be carried over to the next one.